app/src/main/java/com/arygm/quickfix/ui/elements/QuiickFixServicesChoosing.kt

```diff
@@ -0,0 +1,168 @@
+                      horizontalArrangement =
+                          Arrangement.spacedBy(paddingHorizontal, Alignment.CenterHorizontally)) {
+                        serviceTypes.forEach { service ->
+                          val isSelected = service == selectedService
+                          Text(
+                              text = service,
+                              style = MaterialTheme.typography.bodyMedium,
+                              modifier =
+                                  Modifier.clickable { onServiceSelect(service) }
+                                      .background(
+                                          color =
+                                              if (isSelected) colorScheme.primary
+                                              else colorScheme.secondary,
+                                          shape = RoundedCornerShape(cornerRadius))
+                                      .padding(
+                                          horizontal =
+                                              paddingHorizontal *
+                                                  1.5f, // Extra padding inside each option
+                                          vertical = verticalSpacing)
+                                      .testTag("serviceText_$service"),
+                              color =
+                                  if (isSelected) colorScheme.onPrimary
+                                  else colorScheme.onSecondary)
+                        }
+                      }
+
+                  Spacer(
+                      modifier =
+                          Modifier.height(verticalSpacing * 1.5f)) // Space before Apply button
+
+                  // Apply button to confirm the selection
+                  Button(
+                      onClick = onApplyClick,
+                      modifier =
+                          Modifier.fillMaxWidth()
+                              .padding(horizontal = buttonPaddingHorizontal)
+                              .testTag("applyButton"),
+                      colors =
+                          androidx.compose.material3.ButtonDefaults.buttonColors(
+                              containerColor = colorScheme.primary,
+                              contentColor = colorScheme.onPrimary)) {
+                        Text("Apply")
+                      }
+
+                  Spacer(
+                      modifier = Modifier.height(verticalSpacing)) // Space between Apply and Reset
+
+                  // Reset option to clear the selection
+                  Text(
+                      text = "Reset",
+                      color = colorScheme.onSecondaryContainer,
+                      modifier =
+                          Modifier.clickable { onResetClick() }
+                              .padding(vertical = verticalSpacing / 2)
+                              .testTag("resetButton"))
+                }
+          }
+        }
+  }
+}
+
+/**
```

Comment:
Thanks for explnation. The whole code is commented allowing all the team to understand the purpose of your code. Nice job !